client/src/core-components/form.js

```diff
@@ -39,7 +39,13 @@ class Form extends React.Component {
     }
 
     componentDidMount() {
-        this.setState(this.getInitialFormAndValidations());
+        this.setState(this.getInitialFormAndValidations(this.props.children));
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if(nextProps.values && this.props.values && (nextProps.values.template !== this.props.values.template)) {
+            this.setState(this.getInitialFormAndValidations(nextProps.children));
+        }
```

Comment:
This one is deprecated in React, let's use `componentDidUpdate` instead, and let's avoid accessing `.template`, which we don't know if it exists (if we were using TS, this would be an error).